tests/test_py35/test_resp.py

```diff
@@ -96,3 +96,30 @@ def test_ctx_manager_is_coroutine():
     with aiohttp.ClientSession(loop=loop) as session:
         async with await session.post(server.make_url('/'), data=data) as resp:
             assert resp.status == 200
+
+
+async def test_iter_error_on_conn_close(test_server, loop):
```

Comment:
Since you change is not Python 3.5 specific and could be executed by Python 3.4 the test should be placed in `tests/test_client_functional.py` and written with old-styled `asyncio.coroutine` and `yield from`.